recipe/meta.yaml

```diff
@@ -1,52 +1,67 @@
-{% set version = "6.2.1" %}
+{% set name = "gmp" %}
+{% set version = "6.3.0" %}
+{% set posix = 'm2-' if win else '' %}
+{% set am_version = "1.15" %} #  keep synchronized with build.sh
 
 package:
-  name: gmp
+  name: {{ name }}
   version: {{ version }}
 
 source:
-  url: https://gmplib.org/download/gmp/gmp-{{ version }}.tar.bz2
-  sha256: eae9326beb4158c386e39a356818031bd28f3124cf915f8c5b1dc4c7a36b4d7c
-  patches:
-    - 0000-Avoid-the-x18-register.patch
+  url: https://gmplib.org/download/{{ name }}/{{ name }}-{{ version }}.tar.bz2
+  sha256: ac28211a7cfb609bae2e2c8d6058d66c8fe96434f740cf6fe2e47b000d1c20cb
 
 build:
-  number: 3
-  skip: true  # [win]
+  number: 0
   run_exports:
     # excellent, even across major ver:
     #  https://abi-laboratory.pro/tracker/timeline/gmp/
     - {{ pin_subpackage("gmp") }}
 
 requirements:
   build:
-    - gnuconfig  # [unix]
-    - {{ compiler('c') }}
-    - {{ compiler('cxx') }}
-    - m4
-    - make
-    - patch  # [unix]
-  host:
+    - {{ compiler('c') }}                 # [unix]
+    - {{ compiler('cxx') }}               # [unix]
+    - gnuconfig                           # [unix]
+    - m4                                  # [unix]
+    - make                                # [unix]
+    - m2w64-toolchain                     # [win]
+    # m2-autoconf includes m2-m4 for win
+    - {{posix}}autoconf                   # [win]
+    - {{posix}}automake-wrapper           # [win]
+    - {{posix}}automake{{ am_version }}   # [win]
+    - {{posix}}libtool                    # [win]
+    - posix                               # [win]
+    - {{posix}}pkg-config
 
```

Comment:
For Windows, I think we have a runtime dependency on `{{native}}gcc-libs` for `libgmpxx-4.dll`:

```
>dumpbin /dependents gmp\bin\libgmpxx-4.dll
Microsoft (R) COFF/PE Dumper Version 14.16.27051.0
Copyright (C) Microsoft Corporation.  All rights reserved.


Dump of file gmp\bin\libgmpxx-4.dll

File Type: DLL

  Image has the following dependencies:

    libgmp-10.dll
    KERNEL32.dll
    msvcrt.dll
    libgcc_s_seh-1.dll
    libstdc++-6.dll
```

`libgmp-10.dll` seems to be OK:

```
>dumpbin /dependents gmp\bin\libgmp-10.dll
Microsoft (R) COFF/PE Dumper Version 14.16.27051.0
Copyright (C) Microsoft Corporation.  All rights reserved.


Dump of file gmp\bin\libgmp-10.dll

File Type: DLL

  Image has the following dependencies:

    KERNEL32.dll
    msvcrt.dll
```

Reply:
The main concern is that the runtime will be polluted with:
```
    m2w64-gcc-libgfortran:    5.3.0-6                     defaults
    m2w64-gcc-libs:           5.3.0-7                     defaults
    m2w64-gcc-libs-core:      5.3.0-7                     defaults
    m2w64-gmp:                6.1.0-2                     defaults
    m2w64-libwinpthread-git:  5.0.0.4634.697f757-2        defaults
```
and linked incorrectly:
```
   INFO: sysroot: 'C:/Windows/' files: '['winhlp32.exe', 'win.ini', 'twain_32/wiatwain.ds', 'twain_32.dll']'
WARNING (r-gmp,Lib/R/library/gmp/libs/x64/gmp.dll): $RPATH/R.dll not found in packages, sysroot(s) nor the missing_dso_whitelist.
.. is this binary repackaging?
WARNING (r-gmp,Lib/R/library/gmp/libs/x64/gmp.dll): Needed DSO Library/mingw-w64/bin/libgcc_s_seh-1.dll found in ['defaults/win-64::m2w64-gcc-libs-core==5.3.0=7']
WARNING (r-gmp,Lib/R/library/gmp/libs/x64/gmp.dll): .. but ['defaults/win-64::m2w64-gcc-libs-core==5.3.0=7'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
WARNING (r-gmp,Lib/R/library/gmp/libs/x64/gmp.dll): Needed DSO Library/mingw-w64/bin/libgmp-10.dll found in ['defaults/win-64::m2w64-gmp==6.1.0=2']
WARNING (r-gmp,Lib/R/library/gmp/libs/x64/gmp.dll): .. but ['defaults/win-64::m2w64-gmp==6.1.0=2'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
   INFO (r-gmp,Lib/R/library/gmp/libs/x64/gmp.dll): Needed DSO C:/Windows/System32/KERNEL32.dll found in $SYSROOT
   INFO (r-gmp,Lib/R/library/gmp/libs/x64/gmp.dll): Needed DSO C:/Windows/System32/downlevel/api-ms-win-crt-convert-l1-1-0.dll found in $SYSROOT
   INFO (r-gmp,Lib/R/library/gmp/libs/x64/gmp.dll): Needed DSO C:/Windows/System32/downlevel/api-ms-win-crt-environment-l1-1-0.dll found in $SYSROOT
   INFO (r-gmp,Lib/R/library/gmp/libs/x64/gmp.dll): Needed DSO C:/Windows/System32/downlevel/api-ms-win-crt-heap-l1-1-0.dll found in $SYSROOT
   INFO (r-gmp,Lib/R/library/gmp/libs/x64/gmp.dll): Needed DSO C:/Windows/System32/downlevel/api-ms-win-crt-locale-l1-1-0.dll found in $SYSROOT
   INFO (r-gmp,Lib/R/library/gmp/libs/x64/gmp.dll): Needed DSO C:/Windows/System32/downlevel/api-ms-win-crt-math-l1-1-0.dll found in $SYSROOT
   INFO (r-gmp,Lib/R/library/gmp/libs/x64/gmp.dll): Needed DSO C:/Windows/System32/downlevel/api-ms-win-crt-private-l1-1-0.dll found in $SYSROOT
   INFO (r-gmp,Lib/R/library/gmp/libs/x64/gmp.dll): Needed DSO C:/Windows/System32/downlevel/api-ms-win-crt-runtime-l1-1-0.dll found in $SYSROOT
   INFO (r-gmp,Lib/R/library/gmp/libs/x64/gmp.dll): Needed DSO C:/Windows/System32/downlevel/api-ms-win-crt-stdio-l1-1-0.dll found in $SYSROOT
   INFO (r-gmp,Lib/R/library/gmp/libs/x64/gmp.dll): Needed DSO C:/Windows/System32/downlevel/api-ms-win-crt-string-l1-1-0.dll found in $SYSROOT
   INFO (r-gmp,Lib/R/library/gmp/libs/x64/gmp.dll): Needed DSO C:/Windows/System32/downlevel/api-ms-win-crt-time-l1-1-0.dll found in $SYSROOT
WARNING (r-gmp,Lib/R/library/gmp/libs/x64/gmp.dll): Needed DSO Library/mingw-w64/bin/libstdc++-6.dll found in ['defaults/win-64::m2w64-gcc-libs==5.3.0=7']
WARNING (r-gmp,Lib/R/library/gmp/libs/x64/gmp.dll): .. but ['defaults/win-64::m2w64-gcc-libs==5.3.0=7'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
WARNING (r-gmp): dso library package skupr/win-64::ucrt64-gcc-libs==13.2.0=4000 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
WARNING (r-gmp): interpreter (R) package local/win-64::r-base==4.3.1=h60f1e59_0 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
WARNING (r-gmp): run-exports library package local/win-64::gmp==6.3.0=hf4a77e7_1 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
```
As a result, creating a new environment:
`conda create -n foo  r-gmp -c local`
the command `R -e "library('gmp')"` will fail.

Reply:
Is it really being linked incorrectly? conda-build says that `Lib/R/library/gmp/libs/x64/gmp.dll` links `Library/mingw-w64/bin/libgcc_s_seh-1.dll`, `Library/mingw-w64/bin/libgmp-10.dll` and `Library/mingw-w64/bin/libstdc++-6.dll` and that these will not be present at runtime (which would basically cause a failure at runtime, unless they are provided via other means).

Reply:
`Lib/R/library/gmp/libs/x64/gmp.dll` links correctly `Library/mingw-w64/bin/libgcc_s_seh-1.dll` but
`Library/mingw-w64/bin/libgmp-10.dll` **doesn't**  link correctly `'defaults/win-64::m2w64-gmp==6.1.0=2'`. We expect that it link to local gmp 8.3.0. This is a mess